hnn_core/tests/test_gui.py

```diff
@@ -422,6 +423,114 @@ def test_gui_adaptive_spectrogram():
     plt.close('all')
 
 
+def test_gui_layer2_dipole():
+    """Test the visualization of layer 2 dipole in the HNNGUI."""
+    gui = HNNGUI()
+    gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    gui.run_button.click()
+    figid = 1
+    figname = f'Figure {figid}'
+    axname = 'ax1'
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'clear')
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'plot')
+    # Check if data is plotted on the axes
+    assert len(gui.viz_manager.figs[figid].axes) == 2
+    # Check default figs have data on their axis
+    assert gui.viz_manager.figs[figid].axes[1].has_data()
+    plt.close('all')
+
+
+def test_gui_layer5_dipole():
```

Comment:
there seems to be redundancy in the tests. Can you [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) the code? Using loops or [fixtures](https://docs.pytest.org/en/6.2.x/fixture.html)

Reply:
this function looks exactly the same as the function above except for some minor differences